modules/application.py

```diff
@@ -349,6 +350,20 @@ def removeAddon(self):
                                              str(self.addonList.item(row, 0).text())),
                                          Qt.QMessageBox.Yes, Qt.QMessageBox.No)
         if answer == Qt.QMessageBox.Yes:
+            settings = Qt.QSettings()
+            parent = "{}/Interface/AddOns".format(str(settings.value(defines.WOW_FOLDER_KEY, defines.WOW_FOLDER_DEFAULT)))
+            contents = os.listdir(parent)
+            addonName =  str(self.addonList.item(row, 0).text())
+            for item in contents:
+                itemDir = "{}/{}".format(parent, item)
+                if os.path.isdir(itemDir) and not item.lower().startswith("blizzard_"):
+                    toc = "{}/{}.toc".format(itemDir, item)
+                    if os.path.exists(toc):
+                        tmp = self.extractAddonMetadataFromTOC(toc)
+                        if re.search(tmp[0], addonName) != None:
+                            rmtree(itemDir)
+            if addonName == "InFlight Taxi Timer":
```

Comment:
Having special exceptions doesn't seem to be a good idea. Maybe have an error pop-up about "couldn't delete addon" instead for all failed deletions?